ext/rbsecp256k1/rbsecp256k1.c

```diff
@@ -1102,36 +1102,57 @@ Context_alloc(VALUE klass)
  *
  * Context initialization should be infrequent as it is an expensive operation.
  *
+ * @param in_random_bytes [String,nil] (Optional) 32 bytes of random data used
+ *   to randomize the context. If omitted then the context remains
+ *   unrandomized. It is recommended that you provide this argument.
  * @return [Secp256k1::Context] 
  * @raise [RuntimeError] if context randomization fails.
  */
 static VALUE
-Context_initialize(VALUE self)
+Context_initialize(int argc, const VALUE* argv, VALUE self)
 {
   Context *context;
   unsigned char *seed32;
   VALUE random_bytes;
+  VALUE opts;
+  static ID kwarg_ids[1];
+
+  random_bytes = Qnil;
+  if (!kwarg_ids[0]) {
+    CONST_ID(kwarg_ids[0], "random_bytes");
+  }
 
   TypedData_Get_Struct(self, Context, &Context_DataType, context);
 
   context->ctx = secp256k1_context_create(
     SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY
   );
 
-  rb_require("securerandom");
-  random_bytes = rb_eval_string("SecureRandom.random_bytes(32)");
-  Check_Type(random_bytes, T_STRING);
-  if (RSTRING_LEN(random_bytes) != 32)
+  // Handle optional second argument containing random bytes to use for
```

Comment:
Flesh out this comment a bit to explain the need to do `rb_scan_args` followed by `rb_get_kwargs`, as well as what the arguments to `rb_scan_args` mean.